libenkf/src/ert_run_context.c

```diff
@@ -86,7 +90,15 @@ stringlist_type * ert_run_context_alloc_runpath_list(const bool_vector_type * ia
 }
 
 
-static ert_run_context_type * ert_run_context_alloc(bool_vector_type * iactive , run_mode_type run_mode , enkf_fs_type * init_fs , enkf_fs_type * result_fs , enkf_fs_type * update_target_fs , int iter) {
+char * ert_run_context_alloc_run_id( ) {
+  int year,month,day,hour,min,sec;
+  time_t now = time( NULL );
+  unsigned int random = util_dev_urandom_seed( );
+  util_set_datetime_values_utc( now , &sec, &min, &hour, &day, &month, &year);
+  return util_alloc_sprintf("%d:%d:%4d-%0d-%02d-%02d-%02d-%02d:%ud" , getpid() , getuid(), year , month , day , hour , min , sec, random);
```

Comment:
I must admit that I get a bad feeling about embedding this information this clearly/nicely in the ID. Not that I have a problem with them being the _source_ of the ID, but when they are so nicely formatted it is easy for someone (e.g. me in the future) to think "I would like to have the uid, ohh it is right there in the ID, let me just extract it" and suddenly the ID is no longer just a id, it is actually a structure containing more information, and we are locked to keep it like that forever, or figure out who (a)buses the field and change them. I would feel better if we hashed or scrambled the data a bit first. 

If we are interested in the pid/uid/local-timestamp we should send them as explicit fields. I have no problems with that, there is no reason to be stingy with the meta-information we send in the jobs-document, as long as it is explicit.